vhd/lib/libvhd.c

```diff
@@ -4047,18 +4050,23 @@ vhd_block_vector_read(vhd_context_t *ctx, vhd_block_vector_t *vec)
 }
 
 /**
- * @vec: block vector to initialize
- * @block: vhd block number
- * @map: optional bitmap of sectors to map (relative to beginning of block)
- * @buf: destination buffer
- * @blk_start: byte offset relative to beginning of block
- * @blk_end: byte offset relative to beginning of block
+ * @param[in]	ctx			VHD context
+ * @param[out]	vec 		block vector to initialize
+ * @param[in]	block		vhd block number
+ * @param[in]	map			optional bitmap of sectors to map (relative to beginning of block)
+ * @param[out]	buf			destination buffer
+ * @param[in]	blk_start	byte offset relative to beginning of block
+ * @param[in]	blk_end		byte offset relative to beginning of block
  *
- * initializes @vec to describe a read into a contiguous buffer
+ * @return 0 on success, -errno on error
+ * 
+ * Initializes \p vec to describe a read into a contiguous buffer
  * of potentially non-contiguous byte ranges in a given vhd block.
- * only sectors with corresponding bits set in @map (if it is not NULL)
+ * 
+ * Only sectors with corresponding bits set in \p map (if it is not NULL)
  * will be mapped; bits corresponding to unmapped sectors will be cleared.
- * first and last sector maps may be smaller than vhd sector size.
+ * 
+ * First and last sector maps may be smaller than vhd sector size.
```

Comment:
The comment on vhd_block_vector_read_allocated_selective' explains why this might happen: if blk_start/blk_end are not VHD sector size aligned.